src/script/lua_api/l_localplayer.cpp

```diff
@@ -430,14 +430,15 @@ int LuaLocalPlayer::l_hud_get_elements(lua_State *L)
 		return 0;
 
 	lua_newtable(L);
-	u32 max = player->getHudIdMax();
-	for (u32 id = 0; id < max; id++) {
-		HudElement *elem = player->getHud(id);
-		if (elem != nullptr) {
-			push_hud_element(L, elem);
-			lua_rawseti(L, -2, id);
+	player->hudApply([&](const std::vector<HudElement*>& hud) {
+		for (std::size_t id = 0; id < hud.size(); ++id) {
```

Comment:
Couldn't you do something like `(for auto elem : hud)` here (same below)?

Reply:
No, I need to get the id.